### PR TITLE
reformat: unique_counts 

### PR DESCRIPTION
**Pull request by Ya2001**
fixes https://github.com/unifyai/ivy/issues/1232
Edited the unique_counts method to make it easier to read and also added some helpful comments and examples to help
new users understand the purpose and the use cases of this method.

### CODE REVIEW

**Comment by Ya2001** [comment]
Hi there,

Do I need to wait for the tests and checks to complete before I can assign my issue in the development section as it is currently not letting me do so.

Thank you,
Akshay

**Comment by Ya2001** [comment]
unique_counts 
#1232

**Comment by Ya2001** [comment]
fixes #1232

**Comment by Ya2001** [comment]
Hi there @Ishticode,

Thank you so much for the review. I have taken your comments into regard and edited the docstring examples accordingly. 
Yes, that was my mistake for repeatedly making references to the class ivy.Array instead of the required function. 

Wanted to ask if the comments are okay now and if so, that I can proceed with adding them to the backend.

Thank you again :))

**Comment by Ya2001** [comment]
Also was just wanting to ask if there is any documentation or videos available to reference before editing the backend as I do not remember viewing any?

**Comment by Ishticode** [comment]
@Ya2001 Hi. You only need to modify and add type hints etc. in the backend. Don't need a docstring there or examples there . The video [Ivy - Reformatting - Coding Tutorial 01: Explanation](https://www.youtube.com/watch?v=fTifKx8GJrg&t=1492s) should be helpful. You can also take a look at the [contributing page](https://lets-unify.ai/ivy/contributing.html) if you like. Hope that helps. : )

**Comment by Ya2001** [comment]
Hi there,

I was curious about what the mention of lambda was in the ToDo list. There that should be the code edited on the backend. 

Thank you,
Please let me know if there are any other changes required :)))

File: ivy/functional/ivy/set.py [hunk 1]
```diff
@@ -144,26 +144,91 @@ def unique_values(
...
+    Tuple([1,2,3,4],[3,1,2,1])
+
+    >>> x = ivy.asarray([1,2,3,4],[2,3,4,5],[3,4,5,6])
+    >>> y = unique_counts(x)
+    >>> print(y)
+    Tuple([1,2,3,4,5,6],[1,2,3,3,2,1])
+
+    With :code: 'ivy.NativeArray' input:
+
+    >>> x = ivy.NativeArray([0.2,0.3,0.4,0.2,1.4,2.3,0.2])
```
Comment (Ishticode) [inline ivy/functional/ivy/set.py:220]: Need ivy.native_array call in the functional form...

File: ivy/functional/ivy/set.py [hunk 2]
```diff
@@ -144,26 +144,91 @@ def unique_values(
...
-        - first element must have the field name values and must be an array containing
-          the unique elements of x. The array must have the same data type as x.
-        - second element must have the field name counts and must be an array containing
-          the number of times each unique element occurs in x. The returned array must
-          have same shape as values and must have the default array index data type.
+    Instance Method Examples
+    ------------------------
 
+    Using :code:`ivy.Array` instance method:
+    >>> x = ivy.Array([0., 1., 3. , 2. , 1. , 0.])
```
Comment (Ishticode) [inline ivy/functional/ivy/set.py:229]: Could u create the array using `ivy.array`

**Review by Ishticode — CHANGES_REQUESTED** [review]

File: ivy/functional/ivy/set.py (hunk 2 above)
Comment (Ya2001) [reply ivy/functional/ivy/set.py]: Perfect just made those changes now.
Sorry that took a while I'm currently flying back to the UK from India :))

**Comment by Ya2001** [comment]
Just finished passing the lint test. Thank you very much for helping me with this task. What is the next step for me to finish this stage of the internship interview. Is there anything I need to do involving upwork?

**Review by Ishticode — APPROVED** [review]
Looking great. It was a pleasure reviewing you PR. I will merge it now. Thankyou for your contribution. : )

**Comment by Ishticode** [comment]
> Just finished passing the lint test. Thank you very much for helping me with this task. What is the next step for me to finish this stage of the internship interview. Is there anything I need to do involving upwork?

May I refer you to the discord server for internship related questions...

**Pull request merged**